### PR TITLE
LibWeb+WebDriver: Implement the Element Click endpoint, and `window.close()`

### PR DESCRIPTION
**Pull request by trflynn89**
These are the functionalities needed to run the `/fetch/metadata/window-open.https.sub.html` WPT test (which now fails some subtests instead of hanging indefinitely). Both of these features exposed a few bugs that also needed to be fixed.

### CODE REVIEW

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 1]
```diff
@@ -2118,26 +2118,35 @@ Messages::WebDriverClient::PrintPageResponse WebDriverConnection::print_page(Jso
...
+// https://w3c.github.io/webdriver/#dfn-set-the-current-browsing-context
+void WebDriverConnection::set_current_browsing_context(Web::HTML::BrowsingContext& browsing_context)
 {
-    TRY(ensure_current_top_level_browsing_context_is_open());
-    return JsonValue {};
+    // 1. Set session's current browsing context to context.
+    m_current_browsing_context = browsing_context;
+
+    // 2. Set the session's current parent browsing context to the parent browsing context of context, if that context
+    //    exists, or null otherwise.
```
Comment (shannonbooth) [inline Userland/Services/WebContent/WebDriverConnection.cpp:2191]: Should this have an `else` to set to `nullptr` for the otherwise case?

Comment (trflynn89) [reply Userland/Services/WebContent/WebDriverConnection.cpp:2191]: Yep!

**Review by shannonbooth — APPROVED** [review]

**Comment by nico** [comment]
(5th commit reverted in #1663)

**Comment by trflynn89** [comment]
(and the `FIXME: Spec-issue` added in the last commit was resolved in #1692)

**Pull request merged**